core/src/main/kotlin/ch/ergon/dope/resolvable/fromable/UseIndex.kt

```diff
@@ -0,0 +1,56 @@
+package ch.ergon.dope.resolvable.fromable
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.DopeQueryManager
+import ch.ergon.dope.resolvable.Resolvable
+import ch.ergon.dope.resolvable.formatIndexToQueryString
+import ch.ergon.dope.resolvable.formatListToQueryStringWithBrackets
+import ch.ergon.dope.resolvable.formatToQueryStringWithSymbol
+import ch.ergon.dope.resolvable.fromable.IndexType.USING_FTS
+import ch.ergon.dope.resolvable.fromable.IndexType.USING_GSI
+
+private const val USE_INDEX = "USE INDEX"
+
+enum class IndexType(val type: String) {
+    USING_GSI("USING GSI"),
+    USING_FTS("USING FTS"),
+}
+
+class IndexReference(
+    private val indexName: String? = null,
+    private val indexType: IndexType? = null,
+) : Resolvable {
+    override fun toDopeQuery(manager: DopeQueryManager) = DopeQuery(formatIndexToQueryString(indexName, indexType?.type), emptyMap())
+}
+
+class UseIndex(
+    val bucket: Bucket,
+    vararg val indexReference: IndexReference,
+) : Joinable, Deletable, Fromable {
+    override fun toDopeQuery(manager: DopeQueryManager): DopeQuery {
+        val bucketDopeQuery = bucket.toDopeQuery(manager)
+        val indexReferenceDopeQueries = indexReference.map { it.toDopeQuery(manager) }
+        return DopeQuery(
+            queryString = formatToQueryStringWithSymbol(
+                bucketDopeQuery.queryString,
+                USE_INDEX,
+                formatListToQueryStringWithBrackets(indexReferenceDopeQueries, seperator = ", ", prefix = "(", postfix = ")"),
+            ),
+            parameters = bucketDopeQuery.parameters + indexReferenceDopeQueries.fold(emptyMap()) { indexReferenceParameters, field ->
+                indexReferenceParameters + field.parameters
+            },
+        )
+    }
+}
+
+fun UseIndex.useIndex(indexName: String) = UseIndex(bucket, *indexReference, IndexReference(indexName))
+
+fun UseIndex.useGsiIndex(indexName: String? = null) = UseIndex(bucket, *indexReference, IndexReference(indexName, USING_GSI))
```

Comment:
Why did you decide against varargs in these functions?

Reply:
I decided to go without varargs, because then the naming of the function would be a bit strange. What do you suggest?